src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,605 @@
+                    always_validation_status,
+                ) = validation.run_validation_sequence_sets(
+                    world=current_world,
+                    eventually_validation_sequence_set=[[eventually_validation_set]],
+                    always_validation_sequence_set=[[]],
+                )
+
+                if not validation.contains_failure(eventually_validation_status):
+                    break
+
+            except queue.Empty as empty:
+                logger.warning("failed to obtain world")
+
+        validation.check_validation(eventually_validation_status)
+
+    def __validateMatchingRobotIds(self, worldstate: WorldState):
+        """Checks whether the robots to be used in this field test matches up with the robots we see on the field
+
+        Args:
+            worldstate (worldState Proto): The worldstate containing the robots to be used in the test
+
+        Raises:
+            Exception: On there existing a robot to be used in the test but is not on the field
+        """
+        logger.info("testing ids matching")
+
+        # validate that ids match with test setup
+        ids_present = True
+        for idx, robot_ids in enumerate(
+            [worldstate.blue_robots, worldstate.yellow_robots]
+        ):
+            team_name = "friendly" if idx == 0 else "enemy"
+            team_robots = (
+                self.friendly_robot_ids_field
+                if idx == 0
+                else self.enemy_robot_ids_field
+            )
+            for robot_id in robot_ids:
+                if robot_id not in team_robots:
+                    logger.warning(
+                        f"robot {id} from the {team_name} team present in test but not on the field"
+                    )
+                    ids_present = False
+
+        if not ids_present:
+            raise Exception("robotIds do not match")
+
+    def __setBallState(self, worldstate: WorldState):
+        """Commands a robot to move the ball to the given position on the field
+
+        Args:
+            worldstate (worldState Proto): proto that contains ball position
+
+        Raises:
+            Exception: On failing to move the ball to the correct position
+        """
+
+        logger.info("starting ball placement")
+
+        # ball placement
+        if worldstate.HasField("ball_state"):
+
+            ball_position = tbots.createPoint(worldstate.ball_state.global_position)
+
+            dribble_tactic = DribbleTactic(
+                dribble_destination=worldstate.ball_state.global_position,
+                allow_excessive_dribbling=True,
+            )
+            move_ball_tactics = AssignedTacticPlayControlParams()
+            move_ball_tactics.assigned_tactics[
+                self.friendly_robot_ids_field[0]
+            ].dribble.CopyFrom(dribble_tactic)
+
+            self.blue_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, move_ball_tactics
+            )
```

Comment:
I wonder if the ball placement during the setup phase of the test should be done using `BallPlacementPlay` instead of `DribbleTactic`. Since I think that would take care of the logic better for how the robots should safely dribble and which robot should do it. @itsarune what do you think?
This could also be part of a separate ticket for validating auto setting up for field tests.

Reply:
This would require us to follow all ball placement rules when setting up the play - I think its simpler to override the dribble tactic, as our goal is simply to get the ball to the point to start the test. I think of plays as more complex than tactics, so I'd rather rely on a tactic when setting up a test because its less likely to break due to complexity. That's how I see it at least.

Reply:
@PhilD71 brings up a good point. If our goal is just to move the ball to a position to start our test, it makes more sense to use tactics since they're less complex and we don't necessarily have to follow BallPlacementPlay rules

Reply:
I see. Though my main concern is that dribble tactic would try to dribble the ball at max speed which isn't ideal since the ball can easily be lost if it's being pulled. I was under the impression that #2889 will have some extra logic in ball placement play which will push the ball to avoid the problem of losing the ball when pulling.